### PR TITLE
allow deleting subproblems

### PR DESCRIPTION
**Pull request by chinook25**
Allow deleting subproblems.

### CODE REVIEW

File: app/js/subProblem/subProblemController.js [hunk 1]
```diff
@@ -147,6 +136,38 @@ define(['lodash', 'clipboard', 'angular'], function(_, Clipboard) {
...
+        templateUrl: './deleteSubproblem.html',
+        controller: 'DeleteSubproblemController',
+        resolve: {
+          subproblem: function() {
+            return $scope.subProblem;
+          },
+          callback: function() {
+            return function() {
+              SubProblemResource.delete($stateParams).$promise.then(function() {
+                var otherSubproblem = _.reject($scope.subProblems, ['id', $scope.subProblem.id])[0];
```
Comment (DanielReid) [inline app/js/subProblem/subProblemController.js]: othersubproblemS I think? In which case it's not consistent with usage of subProblemChanged (which expects a single problem)

File: nightwatch.local.conf.js [hunk 2]
```diff
@@ -26,7 +26,7 @@ module.exports = {
       desiredCapabilities: {
         browserName: 'firefox',
         'moz:firefoxOptions': {
-          args: ['-headless']
+          // args: ['-headless']
```
Comment (DanielReid) [inline nightwatch.local.conf.js]: probably not correct to commit

File: node-backend/subProblemHandler.js [hunk 3]
```diff
@@ -113,10 +112,57 @@ module.exports = function(db) {
...
+  }
+
+  function deleteTransaction(workspaceId, subproblemId, next, client, transactionCallback) {
+    async.waterfall([
+      _.partial(countSubProblems, workspaceId, next),
+      _.partial(deleteSubproblemAction, subproblemId, next)
+    ], transactionCallback);
+  }
+
+  function countSubProblems(workspaceId, next, callback) {
```
Comment (DanielReid) [inline node-backend/subProblemHandler.js]: probably should be called something like `blockIfOnlyOneSubproblem`

File: test/subproblemHandlerSpec.js [hunk 4]
```diff
@@ -279,4 +281,86 @@ describe('the subproblem handler', () => {
...
+      countSubproblemsForWorkspace = sinon.stub(subproblemRepoStub, 'countSubproblemsForWorkspace');
+      utilStub.handleError = chai.spy();
+    });
+
+    afterEach(() => {
+      deleteStub.restore();
+      countSubproblemsForWorkspace.restore();
+    });
+
+    // var expectations = function(subproblem) {
```
Comment (DanielReid) [inline test/subproblemHandlerSpec.js]: leftovers

File: app/js/subProblem/subProblemController.js (hunk 1 above)
Comment (chinook25) [reply app/js/subProblem/subProblemController.js]: [0] at the end

Comment (DanielReid) [reply app/js/subProblem/subProblemController.js]: doh

**Pull request merged**